be/src/storage/size_tiered_compaction_policy.cpp

```diff
@@ -177,7 +177,17 @@ Status SizeTieredCompactionPolicy::_pick_rowsets_to_size_tiered_compact(bool for
     int64_t level_size = -1;
     int64_t total_size = 0;
     int64_t prev_end_version = -1;
+    bool skip_dup_large_base_rowset = true;
     for (auto rowset : candidate_rowsets) {
+        // when duplicate key's base rowset larger than 0.8 * max_segment_file_size, we don't need compact it
+        if (keys_type == KeysType::DUP_KEYS && skip_dup_large_base_rowset &&
```

Comment:
It seems that only the first rowset whose size over threshold would be skipped, is it by design ?